pep-0497-externals.txt

```diff
@@ -0,0 +1,359 @@
+PEP: 497
+Title: External and environment-specific dependencies
+Version: $Revision$
+Last-Modified: $Date$
+Author: Tennessee Leeuwenburg <tleeuwenburg@gmail.com>,
+
+Status: Draft
+Type: Standards Track
+Python-Version 3.6
+Content-Type: text/x-rst
+Created: 03-Aug-2015
+Post-History: 03-Aug-2015
+
+
+Abstract
+========
+
+The canonical method for installing Python packages is to use pip, whether that
+is referencing PyPI or a local repository. A limitation of the current install
+ecosystem is the relationship with non-Python dependencies. This
+can result in compilation failures or execution failures. Package maintainers
+may wish to specify the requirements to avoid this issue, and users may
+appreciate earlier and more informative messages about their system.
+
+The responsibility of pip is to stop early and provide useful information about
+missing (and not automatically installable) dependencies. Ideally it would provide
+a useful user message about how to obtain the required dependencies. The
+responsibility for actual installation lies with the user at this time.
+
+This PEP proposes a mechanism for extending the requirements section of setup.py
```

Comment:
This is a topic switch from pip to setuptools/distutils. This PEP needs to be clear it covers two things:

1. How users specify external dependencies when using setuptools
1. And how they specify them when using pip (e.g. in requirements.txt files)
2. How external dependencies will be communicated between tools (setuptools -> wheel, wheel -> pip, setuptools -> pip, ...)